### PR TITLE
Feature/410 submodel service implementation

### PR DESCRIPTION
**Pull request by ds-ext-sceronik**
resolves eclipse-tractusx/traceability-foss#410

### CODE REVIEW

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/config/ApplicationStartupConfig.java [hunk 1]
```diff
@@ -80,6 +82,10 @@ public void createNotificationContracts() {
             log.info("on ApplicationReadyEvent notification contracts created.");
         });
         executor.shutdown();
+
+        log.info("test submodelClient on startup");
+        submodelTestOnStartup.createSubmodel("testId", "testPayload");
```
Comment (ds-mwesener) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/config/ApplicationStartupConfig.java:87]: This should be removed.

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/config/FeignGlobalConfig.java [hunk 2]
```diff
@@ -31,6 +31,7 @@
 @EnableFeignClients(basePackages = {
         "org.eclipse.tractusx.traceability.assets.infrastructure.base.irs",
         "org.eclipse.tractusx.traceability.discovery.infrastructure.repository",
+        "org.eclipse.tractusx.traceability.submodel.infrastructure.reposotory"
```
Comment (ds-mwesener) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/config/FeignGlobalConfig.java:34]: typo

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/submodel/infrastructure/reposotory/SubmodelServerClient.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+package org.eclipse.tractusx.traceability.submodel.infrastructure.reposotory;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class SubmodelServerClient {
+
+    private final SubmodelServerApiClient submodelServerApiClient;
+
+    public SubmodelServerClient(SubmodelServerApiClient submodelServerApiClient) {
```
Comment (ds-mwesener) [inline tx-backend/src/main/java/org/eclipse/tractusx/traceability/submodel/infrastructure/reposotory/SubmodelServerClient.java:29]: Please use @RequiredArgsConstructor

**Review by ds-mwesener — CHANGES_REQUESTED** [review]
Please fix issues.

File: tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/config/FeignGlobalConfig.java (hunk 2 above)
Comment (ds-ext-sceronik) [reply tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/config/FeignGlobalConfig.java:34]: indeed :D package was named wrong also :) thanks for noticing !

**Review by ds-mwesener — APPROVED** [review]

**Pull request merged**